src/Microsoft.DotNet.GenFacades/NotSupportedAssemblyGenerator.cs

```diff
@@ -0,0 +1,153 @@
+           return !Log.HasLoggedErrors;
+        }
+
+        public static void GenerateNotSupportedAssemblyFiles(IEnumerable<string> sourceFiles, string intermediatePath, string message, string apiExclusionListPath)
+        {
+            foreach (string sourceFile in sourceFiles)
+            {
+                if (string.IsNullOrEmpty(sourceFile))
+                {
+                    continue;
+                }
+                if (!File.Exists(sourceFile))
+                {
+                    throw new FileNotFoundException($"File {sourceFile} was not found.");
+                }
+                string text = GenerateNotSupportedAssemblyForSourceFile(sourceFile, message, apiExclusionListPath);
+                string path = intermediatePath + Path.GetFileNameWithoutExtension(sourceFile) + ".notSupported.cs";
+                File.WriteAllText(path, text);
+            }
+        }
+
+        public static string GenerateNotSupportedAssemblyForSourceFile(string sourceFile, string Message, string apiExclusionListPath)
+        {
+            string[] apiExclusions;
+            SyntaxTree syntaxTree = CSharpSyntaxTree.ParseText(File.ReadAllText(sourceFile));
+            if (apiExclusionListPath == null)
+            {
+                apiExclusions = null;
+            }
+            else
+            {
+                apiExclusions = File.ReadAllLines(Path.GetFullPath(apiExclusionListPath));
+            }
+            var rewriter = new NotSupportedAssmblyRewriter(Message, apiExclusions);
+            SyntaxNode root = rewriter.Visit(syntaxTree.GetRoot());
+            return root.GetText().ToString();
+        }
+    }
+
+    internal class NotSupportedAssmblyRewriter : CSharpSyntaxRewriter
+    {
+        private string _message;
+        private List<string> _exclusionApis;
+
+        public NotSupportedAssmblyRewriter(string Message, string[] ExclusionApis)
+        {
+            _message = Message;
+            _exclusionApis = ExclusionApis?.ToList();
+        }
+
+        public override SyntaxNode VisitMethodDeclaration(MethodDeclarationSyntax node)
+        {
+            if (node.Body == null)
+                return node;
+
+            if (_exclusionApis != null && _exclusionApis.Contains(GetMethodDefination(node)))
+                return null;
+
+            string message = "{ throw new System.PlatformNotSupportedException(" + $"{ _message }); "+ " }\n";  
+            BlockSyntax block = (BlockSyntax)SyntaxFactory.ParseStatement(message);
+
+            return node.Update(node.AttributeLists,
+                node.Modifiers,
+                node.ReturnType,
+                node.ExplicitInterfaceSpecifier,
+                node.Identifier,
+                node.TypeParameterList,
+                node.ParameterList,
+                node.ConstraintClauses,
+                block,
+                node.SemicolonToken);
```

Comment:
I think we can use the `With*` methods.
```suggestion
            return node.WithBody(block);
```